### PR TITLE
fix: remove uses of __allowUnsafeMonkeyPatching__

### PR DESCRIPTION
**Pull request by erights**
The `__allowUnsafeMonkeyPatching__` option, as the name indicates, is an unsafe temporary hack that we did to work around what we had hoped was a temporary problem. To see if the underlying problem went away, this PR removes those vestigial uses to see what breaks.

When tried under CI, the only thing that broke is golangci-lint 
https://github.com/Agoric/agoric-sdk/runs/7754082612?check_suite_focus=true
which seems unlikely to be related. Reviewers, what else needs to be tested to see if this PR actually breaks anything?

It would be a big relief to remove uses of this unsafe hacky option.

### CODE REVIEW

**Comment by erights** [comment]
See https://github.com/endojs/endo/pull/1249

**Review by kriskowal — APPROVED** [review]
Fantastic. @samsiegart how confident are you that we won’t need to put this work-around back?

**Comment by samsiegart** [comment]
> Fantastic. @samsiegart how confident are you that we won’t need to put this work-around back?

I need to test it manually, there's no unit tests that run under `lockdown` in the wallet UI, so I wouldn't expect tests to fail either way. Do we know exactly what changed to make this unnecessary?

**Comment by erights** [comment]
> Do we know exactly what changed to make this unnecessary?

I do not.

**Comment by erights** [comment]
> I need to test it manually...

@samsiegart I will hold this PR until I get your approval. Thanks!

**Review by samsiegart — APPROVED** [review]
Manual testing looks good

**Pull request merged**
